main.nf

```diff
@@ -103,13 +109,28 @@ else {
   is_staging = false
 }
 
+def process_maf_helper(ch_project_id, ch_maf_centers, maf_centers, maf_center_list, params.create_new_maf_db) {
+  if (maf_centers.contains("ALL")) {
+    // Process centers if the process type is "maf_process"
+    ch_createNewMafDb = ch_maf_centers
+      .map { center -> 
+        def createNewMafDb = (center == maf_center_list[0]) ? true : false
+      }
+    return process_maf(ch_project_id, ch_maf_centers, ch_createNewMafDb).collect()
```

Comment:
So in nextflow, they run the processes list out of order so even if the list stays the same, it could process PROV one run first, next run it processes MSK first, etc (from my experience when running it multiple times). I'm not sure if that's something even in our control, to be able to know what order nextflow will process in. 

Here, even if we have the first center with `create_new_maf_db` = `true`, it's not guaranteed it will run the first center in the list first. Does this code capture the actual process order nextflow will run the centers in when deciding to insert the `create_new_maf_db` value?

Reply:
Yes, this is a big concern. It seems the order of job is completely random but in my test script, the first center run first in several test rounds. 
<img width="868" height="324" alt="Screenshot 2025-07-29 at 12 38 06 PM" src="https://github.com/user-attachments/assets/800523c2-d01d-43a8-9324-db16ec66907a" />

Reply:
How many centers are in your test script? I remember it was just a few?

Reply:
Four.  I can extend it to 30 and see how it goes.

Reply:
Yes I think with fewer centers that just means higher probability of it just getting the first center. When I ran staging, it ran with a different center (that wasn't the first center in the list) every time.

Reply:
Yes, you're right. Even though the first center being assigned as job 1, it finished after several centers. I will fix this. 
<img width="1225" height="607" alt="Screenshot 2025-07-29 at 1 12 20 PM" src="https://github.com/user-attachments/assets/62c84572-c5a6-4520-9990-af1c4dc7b9a0" />

Reply:
The new commit fixed this issue:
<img width="1243" height="334" alt="Screenshot 2025-07-29 at 4 09 37 PM" src="https://github.com/user-attachments/assets/012a6d86-cada-4a22-b3ce-df7a2cd7f85f" />